src/System.Buffers.Primitives/System/SpanExtensions.cs

```diff
@@ -38,124 +38,122 @@ public static bool StartsWith<T>(this ReadOnlySpan<T> items, ReadOnlySpan<T> sli
+                return IndexOf(pSearchSpace, buffer.Length, value0, value1, value2);
             }
         }
 
-        public unsafe static int IndexOfVectorized(this ReadOnlySpan<byte> buffer, byte value)
+        public static int SequentialIndexOf(this ReadOnlySpan<byte> span, byte value)
         {
-            fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
-            {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value);
-            }
+            return SequentialIndexOf(ref span.DangerousGetPinnableReference(), value, span.Length);
         }
 
-        public unsafe static int IndexOfVectorized(this ReadOnlySpan<byte> buffer, byte value0, byte value1)
+        public unsafe static int IndexOf(this ReadOnlySpan<byte> buffer, byte value0, byte value1)
         {
             fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
             {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value0, value1);
+                return IndexOf(pSearchSpace, buffer.Length, value0, value1);
             }
         }
 
-        public unsafe static int IndexOfVectorized(this ReadOnlySpan<byte> buffer, byte value0, byte value1, byte value2)
+        public unsafe static int IndexOf(this ReadOnlySpan<byte> buffer, byte value0, byte value1, byte value2)
         {
             fixed (byte* pSearchSpace = &buffer.DangerousGetPinnableReference())
             {
-                return IndexOfVectorized(pSearchSpace, buffer.Length, value0, value1, value2);
+                return IndexOf(pSearchSpace, buffer.Length, value0, value1, value2);
             }
         }
 
-        private static unsafe int IndexOfVectorized(byte* searchSpace, int length, byte value)
+        private static unsafe int SequentialIndexOf(ref byte searchSpace, byte value, int length)
         {
-            var offset = 0;
-            // If length < vector length the jump over Vector dominates the search; as the Vector section is quite chunky
-            // So do an early search and exit
-            if (length < Vector<byte>.Count)
+            Debug.Assert(length >= 0);
+
+            IntPtr index = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
+            while (length >= 8)
             {
-                for (; offset < length; offset++)
-                {
-                    var ch = searchSpace[offset];
-                    if (ch == value)
-                    {
-                        // goto rather than inline return to keep loop body small
-                        goto shortExit;
-                    }
-                }
-                // Not found
-                offset = -1;
-                shortExit:
-                return offset;
+                length -= 8;
+
+                if (value == Unsafe.Add(ref searchSpace, index))
+                    goto Found;
```

Comment:
Sure, but I grabbed the code from corefx and it has goto Found in many places:

https://github.com/dotnet/corefx/blob/482335d774791c7a43a157300242f73442ffcad3/src/System.Memory/src/System/SpanHelpers.T.cs#L59

https://github.com/dotnet/corefx/blob/482335d774791c7a43a157300242f73442ffcad3/src/System.Memory/src/System/SpanHelpers.byte.cs#L61

Should I fix this in both repos?